os/include/tinyara/arch.h

```diff
@@ -2179,6 +2179,10 @@ void up_wdog_keepalive(void);
 
 #endif
 
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
```

Comment:
Another reboot reason prototype is in os/arch/arm/include/reboot_reason.h. I think the same file is right for this.